runtime/src/main/cpp/SingleLockList.hpp

```diff
@@ -31,7 +32,7 @@ class SingleLockList : private Pinned {
         Node(Args... args) noexcept : value(args...) {}
 
         Value value;
-        std::unique_ptr<Node> next;
+        KStdUniquePtr<Node> next;
```

Comment:
Smart ptr in linked list may cause uncontrolled deep recursion on delete attempt so it's a sort of anti-pettern. Consider raw ptr, or prevent recursive destruction somehow

Reply:
I kept smart pointer, but added manual destruction to avoid destructor recursion. This approach is advocated for here: https://youtu.be/JfmTagWcqoE?t=1581